pandas/tests/reshape/concat/test_index.py

```diff
@@ -100,18 +100,28 @@ def test_concat_rename_index(self):
         tm.assert_frame_equal(result, exp)
         assert result.index.names == exp.index.names
 
-    def test_concat_copy_index_series(self, axis):
+    def test_concat_copy_index_series(self, axis, using_copy_on_write):
         # GH 29879
         ser = Series([1, 2])
         comb = concat([ser, ser], axis=axis, copy=True)
-        assert comb.index is not ser.index
+        if not using_copy_on_write or axis in [0, "index"]:
+            assert comb.index is not ser.index
+        else:
+            assert comb.index is ser.index
```

Comment:
Not for this PR since we already do it for the copy=False case as well (and I suppose it is a general issue). But I am not sure the index should be identical here. An index' attributes are still mutable (eg setting the name), so we should maybe start creating new Index objects for the same data in such cases.

Sidenote: we should maybe discuss in general how Index objects fit into CoW, since those don't use blocks (and thus don't participate in the refs?)